foundation_cms/static/js/blocks/product_review_carousel.js

```diff
@@ -0,0 +1,604 @@
+   */
+  ensureOverflow() {
+    const viewport = this.container.clientWidth || window.innerWidth;
+    const target = viewport * PREFILL_MULTIPLIER;
+    const current = this.container.scrollWidth;
+    const adv = Math.max(
+      1,
+      this.groupAdvance || this.computeGroupAdvanceStatic(GROUP_SIZE),
+    );
+    const deficit = Math.max(0, target - current);
+    const groupsNeeded = Math.min(PREFILL_MAX_LOOPS, Math.ceil(deficit / adv));
+    if (groupsNeeded > 0) {
+      const totalCards = groupsNeeded * GROUP_SIZE;
+      const start = this.computeNextStartIndex();
+      this.appendCardsFromStart(start, totalCards);
+    }
+  }
+
+  // ---------- DOM helpers ----------
+
+  /**
+   * Next logical start index based on the last child's data-index.
+   * @returns {number}
+   */
+  computeNextStartIndex() {
+    const len = this.originalCount || 0;
+    if (!len) return 0;
+    const children = this.track.children;
+    const last = children[children.length - 1];
+    const lastIdx = last ? parseInt(last.getAttribute("data-index"), 10) : -1;
+    return Number.isFinite(lastIdx) && lastIdx >= 0 ? (lastIdx + 1) % len : 0;
+  }
+
+  /**
+   * Append `count` cards starting from `start`, wrapping modulo the original set.
+   * @param {number} start
+   * @param {number} count
+   */
+  appendCardsFromStart(start, count) {
+    const len = this.originalCount || 0;
+    if (!len || count <= 0 || !Array.isArray(this.originalNodes)) return;
+    const frag = document.createDocumentFragment();
+    for (let i = 0; i < count; i++) {
+      const idx = (start + i) % len;
+      const node = this.originalNodes[idx].cloneNode(true);
+      node.setAttribute("data-index", String(idx));
+      frag.appendChild(node);
+    }
+    this.track.appendChild(frag);
+  }
+
+  /**
+   * Append `count` cards using the computed next start index.
+   * @param {number} count
+   */
+  appendCards(count) {
+    if (!this.originalCount || count <= 0) return;
+    const start = this.computeNextStartIndex();
+    this.appendCardsFromStart(start, count);
+  }
+
+  /**
+   * Remove the first `groupSize` cards (recycling step).
+   * Always call with GROUP_SIZE (3). Removing a non-multiple of 3 from the start
+   * reindexes remaining elements and breaks the 3-track nth-child offsets.
+   * @param {number} groupSize
+   */
+  removeFirstGroup(groupSize) {
+    for (let i = 0; i < groupSize; i++) {
+      const first = this.track.firstElementChild;
+      if (!first) break;
+      first.remove();
+    }
+  }
+
+  // ---------- Animation loop ----------
+
+  /**
+   * rAF loop:
+   *  - Convert elapsed time to pixel delta
+   *  - When `next` >= `groupAdvance`, append next GROUP_SIZE, subtract threshold,
+   *    and remove first GROUP_SIZE (preserves :nth-child(3n+*) cadence)
+   *  - Apply integer pixels via scrollLeft, fractional via translate3d()
+   * @param {DOMHighResTimeStamp} [ts]
+   */
+  tick(ts) {
+    if (!this.enabled) return;
```

Comment:
Small suggestion - consider adding a track guard here: 

```suggestion
    if (!this.enabled || !this.track) return;
```
This would prevent potential crashes if `tick()` gets called during destroy/disable transitions.